File: frontend/src/app/models/policy/plan.ts
export interface PlanDto {
    id: string; // Guid
    name: string;
    description: string;
    benefits: string;
    premiumAmount: number;
    coverageAmount: number;
    durationInMonths: number;
    paymentFrequency: string;
    planType: string;
    isActive?: boolean;
}

export interface CreatePlanDto {
    name: string;
    description: string;
    benefits: string;
    premiumAmount: number;
    coverageAmount: number;
    durationInMonths: number;
    paymentFrequency: string;
    planType: string;
}

File: backend/Domain/Entities/Plan.cs
using System.ComponentModel.DataAnnotations;
using Domain.Enums;

namespace Domain.Entities
{
    public class Plan
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required, StringLength(200)]
        public string Name { get; set; } = null!;

        [Required]
        public string Description { get; set; } = null!;

        [Required]
        public decimal PremiumAmount { get; set; }

        [Required]
        public decimal CoverageAmount { get; set; }

        [Required]
        public int DurationInMonths { get; set; }

        [Required]
        public string PaymentFrequency { get; set; } = null!;

        [Required]
        public string PlanType { get; set; } = null!;

        public bool IsActive { get; set; } = true;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ICollection<Policy> Policies { get; set; } = new List<Policy>();
    }
}